kosat-core/src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -172,16 +169,21 @@ class CDCL {
                 }
             }
 
-            else -> attachClause(clause)
+            else -> attachClause(clause, addToDrat = false)
```

Comment:
This line requires a comment, since other two branches have them.
Also, explain why do we need "addToDrat = false" here?

Reply:
This is actually a workaround, and a pretty ugly one. I don't like how current `dratChecker` interact with the solver at all, but to fix this, the DRAT builder needs to be passed as the constructor. I can just remove this parameter, it won't hurt, but for now I want to keep invariant "original clauses are not in the proof".

Reply:
Why not just check `if (clause.learnt) dratBuilder.addClause(clause)`?

Reply:
I did that initially, but it turns out it only makes things difficult. During preprocessing we often need to modify non-learnt clauses, which means it has to be recorded in the proof. However, we cannot make new clause learnt because it is irredundant, so non-learnts have to be added to the proof manually. Even worse, this means that there is no way to distinguish irredundant and initially given clauses later, so those added irredundant clauses are permanently staying in the proof. I can either store separate flags in `Clause` for `learnt` (redundant) and `added` (not given in the CNF initailly), or just duplicate the initial CNF in proof. I find the second approach favourable because it removes the complexity of clause state entirely, but one may argue that it is more "hacky" and requires solver to have `dratBuilder` during every call to `newClause` (which is not the case right now because CNF can be provided in the constructor), adding some complexity.